### PR TITLE
Rename rust elf binary to "app.elf" after build.

### PR DESCRIPTION
**Pull request by agrojean-ledger**

### CODE REVIEW

File: .github/workflows/reusable_build.yml [hunk 1]
```diff
@@ -90,9 +90,12 @@ jobs:
...
           if [[ "${{ needs.call_get_app_metadata.outputs.is_rust }}" == "true" ]];
           then
-              DEVICE_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanosplus/')" && \
+              BUILD_DEVICE_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanosplus/')" && \
+              BIN_DIR_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanos2/')" && \
               cd ${{ needs.call_get_app_metadata.outputs.build_directory }} && \
-              cargo ledger build ${DEVICE_NAME} -- -Zunstable-options --out-dir=./build/${{ matrix.device }}/bin/
+              cargo ledger build ${BUILD_DEVICE_NAME} -- -Zunstable-options --out-dir=./build/${BIN_DIR_NAME}/bin/
+              APP_NAME=$(grep -E '^\s*name\s*=\s*' Cargo.toml | awk -F'["]' '{print $2}')
+              mv ./build/${BIN_DIR_NAME}/bin/${APP_NAME} ./build/${BIN_DIR_NAME}/bin/app.elf
```
Comment (xchapron-ledger) [inline .github/workflows/reusable_build.yml:98]: Would be great if we can remove this part at some point! But let's keep it for another time.

**Review by xchapron-ledger — APPROVED** [review]

File: .github/workflows/_check_app_load_params.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
         with:
           repository: LedgerHQ/ledger-app-database
           path: ./ledger-app-database
-          ref: main
+          ref: update-rust-boilerplate-data
```
Comment (xchapron-ledger) [inline .github/workflows/_check_app_load_params.yml:33]: To be changed before merging

**Review by tdejoigny-ledger — DISMISSED** [review]

File: .github/workflows/_check_app_load_params.yml (hunk 2 above)
Comment (agrojean-ledger) [reply .github/workflows/_check_app_load_params.yml]: Done.

**Review by xchapron-ledger — APPROVED** [review]

File: .github/workflows/reusable_lint.yml [hunk 3]
```diff
@@ -34,3 +42,19 @@ jobs:
           source: ${{ inputs.source }}
           extensions: ${{ inputs.extensions }}
           clangFormatVersion: ${{ inputs.version }}
+
+  job_lint_rust:
+    name: Lint check for Rust app
+    needs: call_get_app_metadata
+    runs-on: ubuntu-latest
+    container:
```
Comment (xchapron-ledger) [inline .github/workflows/reusable_lint.yml:50]:
```suggestion
    if: ${{ needs.call_get_app_metadata.outputs.is_rust == 'true'}}
    container:
```

**Review by xchapron-ledger — APPROVED** [review]

**Pull request merged**